maplibre/src/style/style.rs

```diff
@@ -140,12 +270,30 @@ impl Default for Style {
                     paint: Some(LayerPaint::Raster(RasterLayer::default())),
                     source: None,
                     source_layer: Some("raster".to_string()),
+                    typ: Some("raster".to_string()),
                 },
             ],
         }
     }
 }
 
+#[derive(Debug)]
+pub enum StyleModuleError {
+    Io(io::Error),
+    Json(serde_json::Error),
+}
```

Comment:
Use here https://crates.io/crates/thiserror

We might not want to pass through serde_json here so we can later more easily replace serde with something else.